CHANGELOG.md

```diff
@@ -11,10 +11,16 @@ For details about compatibility between different releases, see the **Commitment
 
 ### Added
 
+- Add the attributes field to all ApplicationUp messages.
+- Add the locations, version_ids, network_ids fields to ApplicationUp messages that were missing them, i.e. ApplicationJoinAccept, ApplicationDownlink, ApplicationDownlinkFailed, ApplicationInvalidatedDownlinks, ApplicationServiceData.
+- Add Timeout and Cache fields in the EndDeviceMetadataStorageConfig of the AS.
```

Comment:
- Add end device attributes to ApplicationUp messages.
  - Add missing fields to  `ApplicationUp` messages (note the indentation, you can list the messages here)
  - Add timeouts and caching to `EndDeviceMetadataStorageConfig` in the Application Server.

Reply:
done